### PR TITLE
lnwire: add length validation in NewSigFromRawSignature

### LINKED ISSUE #6310: repeatedly calling verifymessage-2 causes LND to crash
### Background

Testing LNDg from @cryptosharks131
Originally reported in:
https://github.com/cryptosharks131/lndg/issues/50

### Your environment

RPi4 8GB + SSD
```
₿ uname -a
Linux raspberrypi 5.10.63-v8+ #1496 SMP PREEMPT aarch64 GNU/Linux
₿ python --version
Python 3.7.3
₿ /home/lndg/lndg/.venv/bin/pip --version
pip 22.0.3 from /home/lndg/lndg/.venv/lib/python3.7/site-packages/pip (python 3.7)
₿ lnd --version
lnd version 0.14.2-beta commit=v0.14.2-beta
```
### Steps to reproduce

Pass invoices to be validated with: https://github.com/cryptosharks131/lndg/blob/27cf13d551152abd6d70aad5a08e0b0224bf2920/jobs.py#L106

using: https://api.lightning.community/?python#verifymessage-2

### Expected behaviour
Python script returning data to it's database.

### Actual behaviour
LND crashes:
```
 lnd[8483]: panic: runtime error: index out of range [108] with length 104
 lnd[8483]: goroutine 20187632 [running]:
 lnd[8483]: github.com/lightningnetwork/lnd/lnwire.NewSigFromRawSignature({0x4012c55ce0, 0x68, 0x70})
 lnd[8483]:         github.com/lightningnetwork/lnd/lnwire/signature.go:32 +0x470
 lnd[8483]: github.com/lightningnetwork/lnd/lnrpc/signrpc.(*Server).VerifyMessage(0x402b6e80d0, {0x1a545b8, 0x40057cc390}, 0x400c196b60)
 lnd[8483]:         github.com/lightningnetwork/lnd/lnrpc/signrpc/signer_server.go:524 +0x138
 lnd[8483]: github.com/lightningnetwork/lnd/lnrpc/signrpc._Signer_VerifyMessage_Handler.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/lightningnetwork/lnd/lnrpc/signrpc/signer_grpc.pb.go:274 +0x7c
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x40139b6b88)
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0x8c
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x58
 lnd[8483]: github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).middlewareUnaryServerInterceptor.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x400a287dc0)
 lnd[8483]:         github.com/lightningnetwork/lnd/rpcperms/interceptor.go:790 +0x24c
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x58
 lnd[8483]: github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).MacaroonUnaryServerInterceptor.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x400a287de0)
 lnd[8483]:         github.com/lightningnetwork/lnd/rpcperms/interceptor.go:663 +0xa0
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x58
 lnd[8483]: github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).rpcStateUnaryServerInterceptor.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x400a287e00)
 lnd[8483]:         github.com/lightningnetwork/lnd/rpcperms/interceptor.go:754 +0x90
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x58
 lnd[8483]: github.com/lightningnetwork/lnd/rpcperms.errorLogUnaryServerInterceptor.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x400a287e20)
 lnd[8483]:         github.com/lightningnetwork/lnd/rpcperms/interceptor.go:579 +0x58
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60})
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x58
 lnd[8483]: github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x1a545b8, 0x40057cc390}, {0x15389a0, 0x400c196b60}, 0x400a287da0, 0x40139b6b88)
 lnd[8483]:         github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xc4
 lnd[8483]: github.com/lightningnetwork/lnd/lnrpc/signrpc._Signer_VerifyMessage_Handler({0x154ff40, 0x40007ca8a0}, {0x1a545b8, 0x40057cc390}, 0x400731f560, 0x40007cd170)
 lnd[8483]:         github.com/lightningnetwork/lnd/lnrpc/signrpc/signer_grpc.pb.go:276 +0x14c
 lnd[8483]: google.golang.org/grpc.(*Server).processUnaryRPC(0x40003dae00, {0x1a73238, 0x401b4b2a80}, 0x40095feb40, 0x40005f6150, 0x2801188, 0x0)
 lnd[8483]:         google.golang.org/grpc@v1.38.0/server.go:1286 +0xc14
 lnd[8483]: google.golang.org/grpc.(*Server).handleStream(0x40003dae00, {0x1a73238, 0x401b4b2a80}, 0x40095feb40, 0x0)
 lnd[8483]:         google.golang.org/grpc@v1.38.0/server.go:1609 +0xa54
 lnd[8483]: google.golang.org/grpc.(*Server).serveStreams.func1.2(0x400ab30cd0, 0x40003dae00, {0x1a73238, 0x401b4b2a80}, 0x40095feb40)
 lnd[8483]:         google.golang.org/grpc@v1.38.0/server.go:934 +0x94
 lnd[8483]: created by google.golang.org/grpc.(*Server).serveStreams.func1
 lnd[8483]:         google.golang.org/grpc@v1.38.0/server.go:932 +0x1f0
 systemd[1]: lnd.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
 systemd[1]: lnd.service: Failed with result 'exit-code'.
 systemd[1]: lnd.service: Service RestartSec=30s expired, scheduling restart.
 systemd[1]: lnd.service: Scheduled restart job, restart counter is at 1.
 systemd[1]: Stopped LND on.
 systemd[1]: Starting LND on...
 ```

### PR DESCRIPTION
**Pull request by yyforyongyu**
Fixes #6310

### CODE REVIEW

File: lnwire/signature.go [hunk 1]
```diff
@@ -13,23 +14,58 @@ import (
...
+	//   0x30 <length> 0x02 <length r> r 0x02 <length s> s
+	// which means the length of R is the 4th byte and the length of S is
+	// the second byte after R ends. 0x02 signifies a length-prefixed,
 	// zero-padded, big-endian bigint. 0x30 signifies a DER signature.
 	// See the Serialize() method for btcec.Signature for details.
-	rLen := sig[3]
-	sLen := sig[5+rLen]
+
+	// Reading <length>, remaining: [0x02 <length r> r 0x02 <length s> s]
+	sigLen := sig[1]
```
Comment (guggero) [inline lnwire/signature.go:44]: nit: convert to int here? Also add empty line before comment below.

**Review by guggero — COMMENTED** [review]
Nice fix! Though it looks like some unit tests are failing now.

File: lnwire/signature.go (hunk 1 above)
Comment (yyforyongyu) [reply lnwire/signature.go]: Done!

**Review by guggero — APPROVED** [review]

**Review by Roasbeef — APPROVED** [review]
LGTM 🎨

Needs a rebase!

**Comment by yyforyongyu** [comment]
> LGTM 🎨
> 
> Needs a rebase!

Done!

**Pull request merged**